mathics/builtin/matrices/partmatrix.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+
+"""
+Parts of Matrices
+
+Set of methods for manipulating Matrices.
+"""
+
+
+from mathics.builtin.base import Builtin
+from mathics.builtin.tensors import get_dimensions
+from mathics.core.expression import Expression
+from mathics.core.atoms import Integer
+
+
+class Diagonal(Builtin):
+    """
+    <dl>
+        <dt>'Diagonal[$m$]'
+        <dd>gives a list with the values in the diagonal of the matrix $m$.
+        <dt>'Diagonal[$m$, $k$]'
+        <dd>gives a list with the values in the $k$ diagonal of the matrix $m$.
+    </dl>
+
+    >> Diagonal[{{1, 2, 3}, {4, 5, 6}, {7, 8, 9}}]
+     = {1, 5, 9}
+
+
+
+    >> Diagonal[{{1, 2, 3}, {4, 5, 6}, {7, 8, 9}}, 1]
+     = {2, 6}
+    """
+
+    rules = {
+        "Diagonal[expr_]": "Diagonal[expr, 0]",
+    }
+
+    summary_text = "Return a list with the diagonal elements of a given matrix"
+
+    def apply(self, expr, diag, evaluation):
+        "Diagonal[expr_List, diag_Integer]"
+
+        result = []
+
+        for count, value in enumerate(expr.elements, start=diag.value):
+            if not hasattr(value, "elements") or len(value.elements) <= count:
+                break
+            if count < 0:
+                continue
+            result.append(value.elements[count])
+        return Expression("List", *result)
+
+
+class MatrixQ(Builtin):
+    """
+    <dl>
+    <dt>'MatrixQ[$m$]'
+        <dd>returns 'True' if $m$ is a list of equal-length lists.
+    <dt>'MatrixQ[$m$, $f$]'
+        <dd>only returns 'True' if '$f$[$x$]' returns 'True' for each
+        element $x$ of the matrix $m$.
+    </dl>
+
+    >> MatrixQ[{{1, 3}, {4.0, 3/2}}, NumberQ]
+     = True
+    """
+
```

Comment:
It is leaking a `summary_text` here, I know it is not your code, but we are trying to improve the code, it is hard to add a summary text to 1000 builtins at once so we are doing it bit by bit.